lxd/auth/driver_openfga.go

```diff
@@ -389,7 +389,20 @@ func (e *embeddedOpenFGA) GetPermissionChecker(ctx context.Context, r *http.Requ
 		})
 	}
 
-	// Perform the request.
+	// There is only one server entity, so no need to do a ListObjects request if the entity type is a server. Instead perform a permission check against
+	// the server URL and return an appropriate PermissionChecker.
+	if entityType == entity.TypeServer {
+		err = e.CheckPermission(r.Context(), r, entity.ServerURL(), entitlement)
+		if err == nil {
+			return allowFunc(true), nil
+		} else if IsDeniedError(err) {
+			return allowFunc(false), nil
+		} else {
+			return nil, fmt.Errorf("Failed to get a permission checker: %w", err)
+		}
+	}
+
+	// Else, perform the request.
```

Comment:
You can probably move this block right up to the top of the function. Apart from that and the static analysis, this looks good :)